simulator/interface/IMessage.ts

```diff
@@ -0,0 +1,15 @@
+export enum MessageType {
+
+    CoreAccess = 0,
+    RunProgress,
+    RoundEnd,
+    TaskCount,
+    CoreInitialise,
+    RoundStart
+}
+
+export interface IMessage {
```

Comment:
This structure is really similar to how redux does it actions on the client side coincidentally!

Reply:
Convenient